src/universe/me.ts

```diff
@@ -0,0 +1,173 @@
+    try {
+      const response = await this.http.getClient()(opts)
+
+      const updatedResource = response?.data?.data?.[0]
+
+      if (this.cachedMeData?.user && updatedResource?.preferences) {
+        this.setCachedMeData({
+          ...this.cachedMeData,
+          preferences: updatedResource.preferences,
+          staff: updatedResource
+        })
+      }
+
+      return updatedResource
+    } catch (error) {
+      throwExceptionFromCommonError(error as ClientError)
+      throw new UniverseMePreferencesError(undefined, { error })
+    }
+  }
+
+  private async getPatchConfig (): Promise<{
+    staffRequestOpts: PatchOpts | null
+    meData: MeData
+  }> {
+    const me = this.cachedMeData ?? await this.fetch()
+
+    return {
+      staffRequestOpts: me.staff.id ? {
+        method: 'PATCH',
+        url: `${this.universe.universeBase}/api/v0/staff/${me.staff.id}`,
+        data: []
+      } : null,
+      meData: me
+    }
+  }
+
+  private setCachedMeData (data?: MeData | null): Universe {
+    if (!data) {
+      this.cachedMeData = undefined
+    } else {
+      this.cachedMeData = Object.assign({}, data)
+    }
+
+    return this.universe
+  }
+
+  /**
+   * Fetch the data of the current user.
+   * If you receive an instance of UniverseUnauthenticatedError you should logout the current session and create a new one.
+   * UniverseUnauthenticatedError: 401
+   * UniverseForbiddenError: 403
+   * UniverseBadGatewayError: 502
+   * UniverseServiceUnavailableError: 503
+   * UniverseTimeoutError: 504
+   */
+  public async fetch (skipInterceptors = false): Promise<MeData> {
+    try {
+      const opts = {
+        method: 'GET',
+        url: `${this.universe.universeBase}/api/v0/me`,
+        skipInterceptors
+      }
+
+      const response = await this.http.getClient()(opts)
+
+      this.setCachedMeData(response?.data?.data)
+
+      return response?.data?.data
+    } catch (error) {
+      throwExceptionFromCommonError(error as ClientError)
+      throw new UniverseMeError(undefined, { error })
+    }
+  }
+
+  /**
+   * Fetch the session data of the current user.
+   * If you receive an instance of UniverseUnauthenticatedError you should logout the current session and create a new one.
+   * UniverseUnauthenticatedError: 401
+   * UniverseForbiddenError: 403
+   * UniverseBadGatewayError: 502
+   * UniverseServiceUnavailableError: 503
+   * UniverseTimeoutError: 504
+   */
+  public async session (): Promise<MeData | never> {
+    try {
+      const opts = {
+        method: 'GET',
+        url: `${this.universe.universeBase}/api/v0/me/session`
+      }
+
+      const response = await this.http.getClient()(opts)
+
+      return response?.data?.data
+    } catch (error) {
+      throwExceptionFromCommonError(error as ClientError)
+
+      throw new UniverseSessionError(undefined, { error })
+    }
+  }
+
+  public async addTemplateFavorite (favorite: addTemplateFavoriteArg): Promise<staff.StaffRawPayload> {
```

Comment:
Oh an here we need to return `Promise<MeData>`, bc after adding a template I need to update the `me` data in the store. We don't have to `fetch`, just update the cached preferences I and return the `MeData`.